components/raftstore/src/store/peer.rs

```diff
@@ -3987,9 +3976,20 @@ where
             }
         }
 
-        if self.raft_group.raft.has_pending_conf()
-            || self.raft_group.raft.pending_conf_index > index
-        {
+        // It's safe to transfer leader to a target peer that has already applied the
+        // configuration change, even if the current leader has not yet applied
+        // it. For more details, refer to the issue at:
+        // https://github.com/tikv/tikv/issues/17363#issuecomment-2404227253.
+        if self.raft_group.raft.pending_conf_index > index {
+            info!(
+                "not ready to transfer leader; target peer has an unapplied conf change";
+                "region_id" => self.region_id,
+                "store_id" => self.peer.get_store_id(),
+                "peer_id" => peer_id,
```

Comment:
```suggestion
                "target_peer_id" => peer_id,
```
"peer_id" mostly refer to the current peer.

Reply:
done